File: day_one/part_one.py
def calculate(s: str, goal: int) -> int:
    nums = []
    with open(s, 'r') as f:
        for line in f.readlines():
            line = int(line)
            nums.append(line)
    for n in nums:
        if goal - n in nums:
            return (goal - n) * n
    return 0

if __name__ == "__main__":
    print(calculate('./blob.txt', 2020))

File: day_one/go/part_one.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func valInList(num int, list []int) bool {
	for _, b := range list {
		if b == num {
			return true
		}
	}
	return false
}

func checkList() int {
	var strNums []string
	var intNums []int

	f, err := os.Open("./blob.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		strNums = append(strNums, scanner.Text())
	}

	for _, i := range strNums {
		s, err := strconv.Atoi(i)
		if err != nil {
			panic(err)
		}
		intNums = append(intNums, s)
	}

	for _, n := range intNums {
		if valInList(n, intNums) && valInList(2020 - n, intNums) {
			return (2020 - n) * n
		}
	}
	return 0
}

func main() {
	fmt.Println(checkList())
}
